reconcile/cna/integration.py

```diff
@@ -57,27 +67,75 @@ def assemble_desired_states(self):
+                if provider.provider != PROVIDER_CNA_EXPERIMENTAL:
+                    continue
                 if not isinstance(provider, NamespaceCNAssetV1):
                     continue
                 for resource in provider.resources or []:
-                    self._desired_states[provider.provisioner.name].add_asset(
-                        asset_factory_from_schema(resource)
+                    asset = asset_factory_from_schema(resource)
+                    self._desired_states[provider.provisioner.name].add_asset(asset)
+
+                    # For now we assume that if an asset is bindable, then it
+                    # always binds to its defining namespace
+                    # TODO: probably this should also be done by passing the required namespace vars
+                    #       to the factory method.
+                    if not asset.bindable():
+                        continue
+                    if not (namespace.cluster.spec and namespace.cluster.spec.q_id):
+                        logging.warning(
+                            "cannot bind asset %s because namespace %s does not have a cluster spec with a cluster id.",
+                            asset,
+                            namespace.name,
+                        )
+                        continue
+                    asset.bindings.add(
+                        Binding(
+                            cluster_id=namespace.cluster.spec.q_id,
+                            namespace=namespace.name,
+                            # For now secret_name is implicit.
+                            secret_name=f"{asset.asset_type()}-{asset.name}",
+                        )
                     )
 
     def assemble_current_states(self):
         self._current_states = defaultdict(State)
+
+        # We fetch all assets from API
         for name, client in self._cna_clients.items():
-            cnas = client.list_assets()
             state = State()
-            state.add_raw_data(cnas)
+            for raw_asset in client.list_assets_for_creator(
+                client.service_account_name()
+            ):
+                try:
+                    asset = asset_factory_from_raw_data(raw_asset)
+                    for binding in client.fetch_bindings_for_asset(asset):
+                        asset.bindings.add(
+                            Binding(
+                                cluster_id=binding.get("cluster_id", ""),
+                                namespace=binding.get("namespace", ""),
+                                secret_name=binding.get("secret_name", ""),
+                            )
+                        )
+                    state.add_asset(asset)
+                except UnknownAssetTypeError as e:
+                    logging.warning(e)
+                except AssetError as e:
+                    # TODO: remember this somehow in the state so we don't try to update/create this asset but skip it instead
```

Comment:
For now should we just fail? Same goes for `UnknownAssetTypeError` because it seems unlikely to happen, but would result in trying to create the resource?

Reply:
that would imply that the integration would fail immediately because some assets in our organization would not validate correctly against the metadata schema of the modules. the reason is the current bug in CNA that does not provide proper input parameter metadata in regards to what is mandatory and what not ([default value null](https://issues.redhat.com/browse/OSDEV-854)).

until this issue is fixed or we can have some finer tenancy in an OCM organisation (e.g. with [labels](https://issues.redhat.com/browse/OSDEV-887)), it would be ok-ish to ignore the error instead of provoking an occasional unrelated fail when someone tests CNA modules in the org outside the context of a-i

Reply:
I trust your judgement here and the `TODO` makes it clear this needs to change. I'm fine with this given the experimental designation.